rfc/template/technical-implementation.md

```diff
@@ -0,0 +1,18 @@
+<!--
+Copyright (c) The OpenTofu Authors
+SPDX-License-Identifier: MPL-2.0
+-->
+
+# RFC-{RFC NUMBER} - Technical implementation
+
+## Technical background
+
+In this section, please explain to the casual reader the technical background of your RFC. This may include explaining OpenTofu-specific terminology since not everyone is well versed in OpenTofu internals. For the best results, assume the reader is not familiar with how OpenTofu works internally.
```

Comment:
Also consider adding new documents in the  ./docs folder for general purpose developer documentation which will help in understanding this and other RFCs